platform/darwin/src/MGLOfflineStorage.h

```diff
@@ -341,6 +342,29 @@ MGL_EXPORT
  */
 @property (nonatomic, readonly) unsigned long long countOfBytesCompleted;
 
+
+#pragma mark - Managing Ambient Cache
+
+
+// JK FINISH DOC
+/*
+ Sets the maximum ambient cache size in megabytes. The default maximum cache size is _ MB. To disable ambient caching, set the maximum ambient cache size to 0. Setting the maximum ambient cache size does not impact the maximum size for offline packs.
+ 
+ This method should be called before using the database.
+ */
+
+- (void)setMaximumAmbientCacheSize:(NSInteger)cacheSize withCallback:(void (^)(NSError *_Nullable error))completion;
```

Comment:
Is there a way to get the maximum ambient cache size? It’s weird to have a setter without a getter. Also, it’s unusual for a setter to be asynchronous – does it need to be asynchronous?

Reply:
`setMaximumAllowedMapboxTiles` does not have a getter, so I'm comfortable with the lack of getter for the maximum ambient cache size. The maximum ambient cache size shouldn't be modified after the database has been created. Not sure of any use cases where a getter would be necessary. I updated the table in the initial PR to reflect this.

I was unsure about making it synchronous, since this should be called before the database is initialized. Could you explain a little more about why it should be synchronous?

Reply:
> The maximum ambient cache size shouldn't be modified after the database has been created.

Is this true? What happens if it is called after - for example halving the size? Is this valid @tmpsantos?

If it is true, then this shouldn't be a setter, and we should find a more appropriate place to set this value.

Reply:
To answer my own question:
> This method should always be called before using the database,
> otherwise the default maximum size will be used.

But, also

> Setting the size to 0 will disable the cache if there is no
> offline region on the database.

Does that mean that disabling the cache depending on whatever conditions is not possible?

Reply:
There is some additional discussion in https://github.com/mapbox/mapbox-gl-native/issues/15025 that may be helpful @julianrex. One of the things I'm working on is making sure the documentation around this is clear.

> Does that mean that disabling the cache depending on whatever conditions is not possible?

If this is set after the offline database has been initialized, the default cache size will be used. So conditionally setting the size after the database has been created wouldn't really work.

Another option could be setting the maximum ambient cache size in the `Info.plist`.

Reply:
Also, just a reminder that this PR is still a _draft_. Feedback is appreciated, but it is not ready for review yet 🐌